src/FormButton/index.tsx

```diff
@@ -11,34 +10,28 @@ import { Button, IButtonProps } from '../index';
  * button that is disabled if the schema doesn't pass validation
  * TODO, add tooltip to explain why its disabled?
  */
-interface IFormButton extends Omit<IButtonProps, 'type' | 'onClick'> {
-  schema?: yup.Schema<any>;
-  data?: any;
-  onClick?: (data: any) => void;
-}
+type OwnProps<T> = {
+  schema?: yup.Schema<T>;
+  data?: T;
+  onClick?: () => void;
+};
+
+type IFormButtonProps<T> = Omit<IButtonProps, 'type' | 'onClick'> & OwnProps<T>;
 
-const FormButton: React.FunctionComponent<IFormButton> = ({
-  schema,
-  data,
-  loading,
-  disabled,
-  onClick = noop,
-  ...buttonProps
-}) => {
-  const errors = useValidateSchema(schema, data);
-  const handleClick = React.useCallback(() => onClick(data), [onClick, data]);
+function FormButton<T>({ schema, data, loading, disabled, onClick, ...buttonProps }: IFormButtonProps<T>) {
+  const [{ errors, isValidating }] = useValidateSchema(schema, data);
 
   return (
     <Button
-      disabled={errors.length > 0 || loading || disabled}
+      disabled={errors.length > 0 || isValidating || loading || disabled}
       loading={loading}
-      onClick={handleClick}
+      onClick={onClick}
```

Comment:
We could leave this out now and use the typings from `IFormButtonProps`. What do you think?

```suggestion
```

Reply:
Ok, but I still need to bind the event handler, so this line has to stay, right?

Reply:
Nope, we can pass it straight through with the other button props being destructured on the line below. Just need to remove it from the react props above and remove the onClick typing: https://github.com/stoplightio/ui-kit/pull/146/files/794831b3eb395472ed38e6129959670f333ebead#diff-d2297309074b8bcc21abf16d7bd22309R21

Main use case is to allow handling the click event

```tsx
<FormButton
  onClick={e => e.preventDefault()}
/>
```

Reply:
Ah I understand what you mean now, ok.